csp_billing_adapter/config.py

```diff
@@ -17,40 +17,68 @@
 import yaml
 
 
-class Config:
+class Config(dict):
+    """The Config class handles reading configuartion settings
+    from a provided configuration file and loading default settings from
+    any configured plugins  This class is based on a dict to allow access
+    using both dot notation and key lookup.
+
+    For example:
+
+        result = Config(config_data)
+        print(result.get('Name'))
+        print(result.Name)
+
+    """
+
     def __init__(self, data: dict):
+        super().__init__(data)
         for key, val in data.items():
             if isinstance(val, (list, tuple, set)):
-                attr = []
-                for item in val:
-                    attr.append(self.parse_value(item))
-
+                attr = [self.parse_value(item) for item in val]
                 setattr(self, key, attr)
             else:
                 setattr(self, key, self.parse_value(val))
 
     @staticmethod
     def parse_value(item):
+        """ recursive evaluation if item is dict """
         return Config(item) if isinstance(item, dict) else item
 
     @staticmethod
     def load_defaults(data, hook):
+        """Load defaults settings/values for any plugins that provide
+        defaults are part of their implementation
+
+        Args:
+            config_path (filepath, optional): The path to the pint-data-manager
+            configuration file. Defaults to
+            '~/.config/pint_data_manager/default.yaml'
+
+        Returns:
+            [dict]: A dictionary containing all of the configuration settings
+
+
+        """
         defaults = {}
         hook.load_defaults(defaults=defaults)
         return {**defaults, **data}
 
     @classmethod
     def load_from_file(cls, filename, hook):
-        with open(filename, 'r') as fh:
+        """Load settings/values from the provided configuration file.
+
+        or any plugins that provide
+        defaults are part of their implementation"""
+
+        with open(filename, 'r', encoding='utf-8') as fh:
             yaml_data = yaml.safe_load(fh)
 
         data = cls.load_defaults(yaml_data, hook)
 
         return cls(data)
 
     def __repr__(self):
```

Comment:
If class inherits from dict then is __repr__ duplicating parent class logic?

Reply:
@smarlowucf God point. I was not sure if you had added this for any other reason. Let me test with and without the __repr__ and if there is no difference, I will remove it.